variant_annotator.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/python3
+
+import csv
+import requests
+import time
+import sys
+
+
+class VariantAnnotator:
+    
+    variant_list = []
+    
+    def __init__(self) -> None:
+        timestr = time.strftime("%Y%m%d-%H%M%S")
+        self.filename_out = 'annotated_data_' + timestr + '.tsv'
+        self.create_output_file()
+        self.read_stdin_clean()
+    
+    def read_stdin_clean(self):
+        # TODO: check for stdin
+        for line in sys.stdin:
+            self.variant_list.append(line.strip())  # Remove trailing whitespace and return chars
+            
+    def get_annotations(self, variant_in) -> dict:    
+        url = 'https://rest.ensembl.org/vep/human/hgvs/' + variant_in
+        response = requests.get(url, headers={"Content-Type" : "application/json"})
+        
+        # Only process successful (200) responses
+        if response.status_code == 200:
+            annotation_json_data = response.json()[0]  # json object is returned in a one item list. just return the zero item for ease of reading
+
+            # Put data in a dict for later processing. Dict is good as we can reference by name later.
+            # TODO: check for missing data? Not sure.
+            annotation_dict = {
+                'variant': variant_in,
+                'assembly_name': annotation_json_data['assembly_name'],
+                'seq_region_name': annotation_json_data['seq_region_name'],
+                'start': annotation_json_data['start'],
+                'end': annotation_json_data['end'],
+                'most_severe_consequence': annotation_json_data['most_severe_consequence'],
+                'strand': annotation_json_data['strand'],
+                'gene_symbol': annotation_json_data['transcript_consequences'][0]['gene_symbol']
+            }
+            # print(variant_in, file=sys.stdout)
+            return annotation_dict
+        else:
+            print('The input value: "' + variant_in + '" failed with a response of: "' + str(response.status_code) + '"', file=sys.stderr)
```

Comment:
Only outputting the status code here, removes a lot of information about the failure. Would prefer to log the error message directly.

Reply:
We generally avoid print statements in favor of logging, so it's more explicit about the level of severity of what's being logged. Also helps keep logging messages isolated from stdout